tests/jerry/array-prototype-tolocalestring.js

```diff
@@ -54,3 +54,33 @@ try {
   assert (e.message === "foo");
   assert (e instanceof ReferenceError);
 }
+
+// Checking behavior when null is passed to the function
+try {
+  Array.prototype.toLocaleString.call(null);
+} catch (e) {
+  assert(e instanceof TypeError);
+}
+
+// Checking behavior when null is passed to the function
+try {
+  var obj = {};
+  Object.defineProperty(obj, 'length', { 'get' : function () { throw new ReferenceError ("foo"); } });
+  Array.prototype.toLocaleString.call(obj);
+  assert(false);
+} catch (e) {
+  assert(e instanceof ReferenceError);
```

Comment:
Please check the message as well! (And use unique messages)

Reply:
The thrown error and the one in the function are not the same `ReferenceError`s, so as I check the message, I get assertion fail.

Reply:
Something is not right then. Q1: Why set the error message then, if it cannot be checked? Q2: Does the error object get replaced somewhere by the builtin implementation? Q3: If yes, why? Q4: If yes, is it necessary to throw a reference error within the anonymous function or any kind of error can be thrown?

Reply:
Sorry I probably made a typo, thats why it did not work. It works with the message check.

Reply:
Sorry I made a typo, that is why it did not work.